### PR TITLE
Add classification script

### PR DESCRIPTION
**Pull request by xinyual**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

cc @dmlc/gluon-nlp-team

### CODE REVIEW

File: scripts/classification/train_classification.py [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+    import horovod.mxnet as hvd
+except ImportError:
+    pass
+from classification import TextPredictionNet
+
+mx.npx.set_np()
+
+_LABEL_PROJ = {'rte':{'entailment':0, 'not_entailment':1},
+               'mnli':{'neutral':0, 'entailment':1, 'contradiction':2}
+               }
```
Comment (sxjscience) [inline scripts/classification/train_classification.py:34]: You may try to determine it automatically.

File: scripts/classification/classification_utils.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+class GlueTask:
+    def __init__(self, task_name, class_num, feature_column, label_column, metric):
+        self.task_name = task_name
+        self.class_num = class_num
+        self.feature_column = feature_column
+        self.label_column = label_column
+        self.metric = metric
+
+
+class SST(GlueTask):
```
Comment (sxjscience) [inline scripts/classification/classification_utils.py:61]: I would recommend to support loading the input as a pandas dataframe.

Comment (xinyual) [reply scripts/classification/classification_utils.py:61]: Ok, I'll change that

**Comment by sxjscience** [comment]
Would you support tensorboard?

File: scripts/classification/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+## some result
+here are some results with their hyperparameters
+
+| task Name    | learning rate  | batch size | seed | epoch | result | tensorboard dev |
+|-----------|---------------|--------------|---------|-------|------|-----|
+|    SST    |    2e-5       | 32    | 7800 |  5 |  92.20 | https://tensorboard.dev/experiment/eKVI0DC6SEWBbHzS8ZphNg/|
+|    STS    |    2e-5       | 32    | 24 |  10 |  88.97 |  https://tensorboard.dev/experiment/kPOnlNeiQ4W5EmFlkqjC6A/|
+|    CoLA    |    2e-5       | 32    | 7800 |  10 |  60.32 |  https://tensorboard.dev/experiment/33euRGh9SrW3p15JWgILnw/ |
```
Comment (sxjscience) [inline scripts/classification/README.md:51]: You may also add the name of the metric.

**Comment by leezu** [comment]
Could you fix this warning in the documentation build so that the CI will pass

> [2021-03-05T00:43:22.211Z] /workspace/gluon-nlp/docs/model_zoo/classification/index.rst: WARNING: document isn't included in any toctree

Essentially you need to link the file in https://github.com/dmlc/gluon-nlp/blob/master/scripts/index.rst

**Comment by leezu** [comment]
The issue `[2021-03-10T11:39:50.280Z] /workspace/gluon-nlp/docs/model_zoo/classification/index.rst: WARNING: document isn't included in any toctree` still remains

**Comment by leezu** [comment]
@barry-jin do you think the issue is related to the CI website-build setup?

**Comment by barry-jin** [comment]
> @barry-jin do you think the issue is related to the CI website-build setup?
Try to add classification/index in toctree
```
.. toctree::
   :hidden:
   :maxdepth: 1

   benchmarks/index
   classification/index
   conversion_toolkits/index
   datasets/index
   generation/index
   machine_translation/index
   pretraining/index
   processing/index
   question_answering/index
```

**Review by leezu — APPROVED** [review]
It may be better to report average result of 3 or 5 random seeds instead of fixed seed. But we can merge this PR for now already. Thank you!
